backend/clap/src/main/java/zpi/ppea/clap/repository/EmailRepository.java

```diff
@@ -0,0 +1,43 @@
+package zpi.ppea.clap.repository;
+
+import data_store.DataStoreGrpc;
+import data_store.EmailRequest;
+import data_store.EmailResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.devh.boot.grpc.client.inject.GrpcClient;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Recover;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Repository;
+import zpi.ppea.clap.exceptions.NoAccessToResource;
+
+import java.util.concurrent.ExecutionException;
+
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+public class EmailRepository {
+
+    @GrpcClient("dataStore")
+    DataStoreGrpc.DataStoreFutureStub dataStoreFutureStub;
+
+    @Retryable(retryFor = NoAccessToResource.class,
```

Comment:
On the second thought maybe it's not a good idea to retry this. If there is an error during grpc transport, and clap asks again for that data, it is possible for data-store to respond with different details:
* The first time clap asks for details, data-store will respond with `let's send first warning`
* This response (data-store --> clap) failed during grpc transport, data-store doesn't know that
* Clap asks for second time, but data-store thinks we already sent email with first warning so it won't respond with same data.

Ideally we would need for Clap after sending emails to make another call to data-store confirming what emails were sent. Only then data-store would acknowledge that how many emails were sent.

But this would require few more lines of implementation on both sides. @Remmo1 @gcfbn What do you think?

Reply:
Yeah, we can go for such confirmation mechanism.